http/stateful-basic/src/main/java/org/wildfly/security/http/sfbasic/BasicMechanismFactory.java

```diff
@@ -52,6 +52,9 @@ public BasicMechanismFactory() {
     public BasicMechanismFactory(final Provider provider) {
     }
 
+    public BasicMechanismFactory(final Provider... provider) {
```

Comment:
Looks like this can be removed since it's not part of the changes needed for this PR.

Reply:
This was added on purpose to do the same that is currently being done by the testing part for the other mechs ([instantiating all of them with providers list](https://github.com/wildfly-security/wildfly-elytron/blob/1.x/tests/base/src/test/java/org/wildfly/security/http/impl/AbstractBaseHttpTest.java#L88-L90)). But it can be removed as basic does not use any provider and can be created with the non-argument constructor. Modified!